### ISSUE TITLE
Another LLVM build fails on MacOS with clang 8.0.0

### ISSUE THREAD

**Issue by jppelteret** [spack/spack#3215]
LLVM (3.9.0, 3.9.1 and trunk) fails to build  on MacOS with clang 8.0.0:
```sh
[  5%] Building C object projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.eprintf.dir/eprintf.c.o
cd /private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/spack-build/projects/compiler-rt/lib/builtins && /Users/jp/Programs/GitHub/Spack/spack/lib/spack/env/clang/clang  -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/spack-build/projects/compiler-rt/lib/builtins -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/projects/compiler-rt/lib/builtins -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/spack-build/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/include  -O3 -DNDEBUG -arch i386    -isysroot -mmacosx-version-min=10.5 -arch i386 -fPIC -O3 -fvisibility=hidden -DVISIBILITY_HIDDEN -Wall -fomit-frame-pointer -o CMakeFiles/clang_rt.eprintf.dir/eprintf.c.o   -c /private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/projects/compiler-rt/lib/builtins/eprintf.c
clang: warning: no such sysroot directory: '-mmacosx-version-min=10.5'
/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-k6HyYw/llvm-3.9.1.src/projects/compiler-rt/lib/builtins/eprintf.c:14:10: fatal error: 'stdio.h' file not found
#include <stdio.h>
         ^
1 error generated.
make[2]: *** [projects/compiler-rt/lib/builtins/CMakeFiles/clang_rt.eprintf.dir/eprintf.c.o] Error 1
```

~~Sadly this issue is not limited to `spack`, e.g.  see [here](https://github.com/Homebrew/homebrew-core/issues/6492), [here](https://github.com/Homebrew/homebrew-core/issues/3030), and [here](https://stackoverflow.com/questions/39438048/clang-warning-no-such-sysroot-directory-mmacosx-version-min-10-5-on-compil). What is puzzling me is that `compiler-rt`, which seems to be the offending subpackage, is built even if I tell `spack` not to do so. This is in my packages.yaml:~~
```
  llvm: # compiler-rt fails with a commonly reported error about stdio.h, due to a bogus parameter being passed as a setting
    version: [3.9.1]
    variants: +all_targets+clang~compiler-rt~debug+gold+internal_unwind+libcxx~link_dylib+lldb+polly~shared_libs
```
and the output of 
```sh
$ spack spec llvm
Input spec
--------------------------------
llvm

Normalized
--------------------------------
llvm
    ^cmake@2.8.12.2:
    ^python@2.7:2.8
        ^bzip2
        ^ncurses
        ^openssl
            ^perl@5.14.0:
            ^zlib
        ^readline
        ^sqlite

Concretized
--------------------------------
llvm@3.9.1%clang@8.0.0-apple+all_targets+clang~compiler-rt~debug+gold+internal_unwind+libcxx~link_dylib+lldb+polly~shared_libs arch=darwin-elcapitan-x86_64
    ^binutils@2.27%clang@8.0.0-apple+gold~krellpatch~libiberty arch=darwin-elcapitan-x86_64
        ^bison@3.0.4%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
            ^m4@1.4.18%clang@8.0.0-apple+sigsegv arch=darwin-elcapitan-x86_64
                ^libsigsegv@2.10%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
        ^flex@2.6.1%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
            ^gettext@0.19.8.1%clang@8.0.0-apple+bzip2+curses+git~libunistring+libxml2+tar+xz arch=darwin-elcapitan-x86_64
                ^bzip2@1.0.6%clang@8.0.0-apple+shared arch=darwin-elcapitan-x86_64
                ^libxml2@2.9.4%clang@8.0.0-apple~python arch=darwin-elcapitan-x86_64
                    ^pkg-config@0.29.1%clang@8.0.0-apple+internal_glib arch=darwin-elcapitan-x86_64
                    ^xz@5.2.3%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
                    ^zlib@1.2.10%clang@8.0.0-apple+pic+shared arch=darwin-elcapitan-x86_64
                ^ncurses@6.0%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
                ^tar@1.29%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
            ^help2man@1.47.4%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
    ^cmake@3.6.1%clang@8.0.0-apple~doc+ncurses+openssl+ownlibs~qt arch=darwin-elcapitan-x86_64
        ^openssl@1.0.2j%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
            ^perl@5.24.0%clang@8.0.0-apple+cpanm arch=darwin-elcapitan-x86_64
    ^libedit@3.1%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
    ^python@2.7.13%clang@8.0.0-apple~tk~ucs4 arch=darwin-elcapitan-x86_64
        ^readline@6.3%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
        ^sqlite@3.8.5%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
    ^swig@3.0.11%clang@8.0.0-apple arch=darwin-elcapitan-x86_64
        ^pcre@8.39%clang@8.0.0-apple+utf arch=darwin-elcapitan-x86_64
```

[Build log attached.](https://github.com/LLNL/spack/files/793758/spack-build.out.txt)

**Comment by davydden** [comment]
I am trying to build it on Sierra at the moment...

**Comment by jppelteret** [comment]
I suppose for full disclosure I recently deleted `XCode.app` via the instructions given in the comments [here](https://apple.stackexchange.com/questions/231915/removing-xcode-while-keeping-the-command-line-tools-on-el-capitan) and now just have the command line tools. I didn't reinstall `spack` afterwards (do I need to)? Perhaps the results of @davydden's build will tell me if this is the root of the problem.

**Comment by jppelteret** [comment]
I was wrong about not being able to disable `compiler-rt`- this issue was fixed in #3015. It seems like the remaining problem likely needs to be fixed upstream? If there's agreement with this then this issue can be closed.

**Comment by jppelteret** [comment]
I've discovered that there's an issue building `lldb` on `MacOS` - the executable needs to be cosigned. Again, this is apparently an [expected issue](https://github.com/Homebrew/legacy-homebrew/pull/46721/files#diff-bd87cd1b624b55e122d22396a7516003R167).

I had to follow [these instructions](https://llvm.org/svn/llvm-project/lldb/trunk/docs/code-signing.txt) in to resolve the issue.

Except from build file:
```
[ 97%] Building CXX object tools/lldb/source/Symbol/CMakeFiles/lldbSymbol.dir/JavaASTContext.cpp.o
cd /private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/spack-build/tools/lldb/source/Symbol && /Users/jp/Programs/GitHub/Spack/spack/lib/spack/env/clang/clang++   -DHAVE_ROUND -DLIBXML2_DEFINED -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/spack-build/tools/lldb/source/Symbol -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/tools/lldb/source/Symbol -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/tools/lldb/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/spack-build/tools/lldb/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/spack-build/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/include -I/Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-8.0.0-apple/python-2.7.13-dlwarog6so2pdr2yajuv5j4c2igdi5um/include/python2.7 -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/tools/clang/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/spack-build/tools/lldb/../clang/include -I/Users/jp/Programs/GitHub/Spack/spack/opt/spack/darwin-elcapitan-x86_64/clang-8.0.0-apple/ncurses-6.0-6njlipmhhkb7vulrx3ftc3af7aryq3h7/include -I/private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/tools/lldb/source/.  -std=c++11  -fPIC -fvisibility-inlines-hidden -Wall -W -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wcovered-switch-default -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Werror=date-time -std=c++11 -Wno-deprecated-declarations -Wno-unknown-pragmas -Wno-strict-aliasing -Wno-deprecated-register -Wno-vla-extension -fPIC -fPIC -O3 -DNDEBUG    -fno-exceptions -o CMakeFiles/lldbSymbol.dir/JavaASTContext.cpp.o -c /private/var/folders/lc/b4bjrtcn69d279m6rf4f42y00000gn/T/jp/spack-stage/spack-stage-ISMF3B/llvm-3.9.1.src/tools/lldb/source/Symbol/JavaASTContext.cpp
lldb_codesign: no identity found
make[2]: *** [bin/debugserver] Error 1
make[2]: *** Deleting file `bin/debugserver'
make[1]: *** [tools/lldb/tools/debugserver/source/MacOSX/CMakeFiles/debugserver.dir/all] Error 2
```
